source/polygon.cpp

```diff
@@ -0,0 +1,46 @@
+ld perimeter(vector<Point<ll>>& poli){
+    ld res = 0.0;
+    int n = ssize(poli);
+    for(int i = 1; i <= n; ++i) res += poli[(i-1) % n].dist(poli[(i)%n]);
+    return res;
+}
+
+// returns 2*area; area < 0 if clock-wise, area > 0 if ccw
+ll double_area(vector<Point<ll>>& poli){
+    ll res = 0.0;
+    int n = ssize(poli);
+    for(int i = 0; i < n; ++i)
+        res += (poli[(i)%n].x * poli[(i+1)%n].y - poli[(i+1)%n].x*poli[(i)%n].y);
+    return res;
+}
+
+bool check_convex(vector<Point<ll>>& poli){
+    int n = ssize(poli);
+    if(n < 3) return false;
```

Comment:
A bit of a technicality, but aren't point and lines convex? We probably shouldn't be working with them as polygons either way, but I think it makes sense to return true here

Reply:
Technically, they are not polygons.
Plus, they are always convex objects so we don't need to check this property for them.
[https://en.wikipedia.org/wiki/Polygon#Naming](url)
![image](https://user-images.githubusercontent.com/72706077/164027342-d8e94079-8dc8-40e7-8ba3-72c0f8e56571.png)

Returning true/false is up to you, let me know.

Reply:
I'd still return true here, so that for base cases we have a little safety net - I agree that we shouldn't be calling this function with 1 or two vertices, but saying that a line or point isn't convex is worse than saying tha it is imho

Reply:
oki